File: Codevita/2019/Round1/Zone 2/SimilarChar/similar.py
# # L = int(input())
# s = "abacsddaa"

# Q = 2

# for _ in range(Q):
#     i = int(input())
#     c = s[i-1]
#     print(s.count(c, 0, i-1))


L = int(input())
s = input()

Q = int(input())

res = []
for _ in range(Q):
    i = int(input())
    c = s[i-1]
    count = 0
    for l in range(i-1):
        if s[l] == c:
            count += 1
    res.append(count)

for i in res[:-1]:
    print(i)
print(res[-1], end='')

File: Codevita/2019/Round1/Zone 2/SimilarChar/similar.java
import java.util.Scanner;

public class similar {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int L = Integer.parseInt(sc.nextLine());
        String s = sc.nextLine();
        int Q = sc.nextInt();

        int[] res = new int[Q];

        for (int i = 0; i < Q; i++) {
            int index = sc.nextInt();
            char ch = s.charAt(index - 1);
            int count = 0;

            for (int j = 0; j < index - 1; j++) {
                if (s.charAt(j) == ch) {
                    count++;
                }
            }
            res[i] = count;
        }
        sc.close();
        for (int i = 0; i < Q-1; i++) {
            System.out.println(res[i]);
        }
        System.out.print(res[Q-1]);
    }
}